src/MergePlugin.php

```diff
@@ -422,6 +429,41 @@ protected function mergeLinks(array $origin, array $merge, array &$dups)
     }
 
     /**
+     * Merge the extras sections of two config files.
+     *
+     * @param RootPackage $root The root package.
+     * @param CompletePackage $package The imported package to merge.
+     */
+    public function mergeExtras(
+        RootPackage $root,
+        CompletePackage $package
+    ) {
+        $this->debug('Merging extra section');
+        $packageExtra = $package->getExtra();
+        if (empty($packageExtra)) {
+            return;
+        }
+
+        if (isset($packageExtra['merge-plugin'])) {
+            $this->debug("Skipping merge-plugin key");
+            unset($packageExtra['merge-plugin']);
+        }
+        $rootExtra = $root->getExtra();
+        foreach ($packageExtra as $key => $value) {
+            $this->debug("Merging extra key <comment>{$key}</comment>");
+            if (isset($rootExtra[$key])) {
```

Comment:
> What was your thinking?

I'm not entirely sure to be honest. The extra section is arbitrary config for arbitrary plugins so the consequences of merging, replacing or otherwise changing it are unclear to me. I don't know for example if configuration for plugin X would even be merged before the plugin attempted to read its settings via `RootPackage::getExtra()`.

Maybe we could start with an `extra.merge-plugin.merge_extra` setting that defaults to false but could be toggled to true in the top level composer.json. When this flag is true, it would default to the basic "first setting wins" strategy with  `extra.merge-pugin` being ignored. If `extra.merge-plugin.replace` is enabled then the last setting would win instead of the first. In the default mode, I would suggest logging when a duplicate key is encountered rather than throwing a fatal exception. This logging could probably be skipped if the current extra key and the key to be merged had identical values.

Would that satisfy your current use case?